src/main/java/OrderDelocalized/OrderService.java

```diff
@@ -3,11 +3,18 @@
 public class OrderService {
 
     public void placeOrder(Customer customer, Order order) {
-
+        customer.addOrder(order);
+        System.out.println("Order has been placed");
     }
 
     public void processOrder(Customer customer, Order order) {
-
+        if (customer.getOpenOrders().contains(order)) {
+            double price = order.calculatePrice();
```

Comment:
(CR: It is not checked if the customer is a member and thus eligible for the discount)